drivers/gpio/gpio_mcux_lpc.c

```diff
@@ -20,82 +19,119 @@
 #include <fsl_common.h>
 #include "gpio_utils.h"
 #include <fsl_gpio.h>
+#include <fsl_pint.h>
+#include <fsl_inputmux.h>
 #include <fsl_device_registers.h>
 
-#define PORT0_IDX	0u
-#define PORT1_IDX	1u
+#define PORT0_IDX 0u
+#define PORT1_IDX 1u
+
+#define PIN_TO_INPUT_MUX_CONNECTION(port, pin) \
+	((PINTSEL0 << PMUX_SHIFT) + (32 * port) + (pin))
+
+#define NO_PINT_INT ((1 << sizeof(pint_pin_int_t)) - 1)
 
 struct gpio_mcux_lpc_config {
 	GPIO_Type *gpio_base;
+	PINT_Type *pint_base;
+	IOCON_Type *pinmux_base;
 	u32_t port_no;
 	clock_ip_name_t clock_ip_name;
 };
 
 struct gpio_mcux_lpc_data {
-	/* gpio_driver_data needs to be first */
-	struct gpio_driver_data common;
+	struct gpio_driver_data general;
 	/* port ISR callback routine address */
 	sys_slist_t callbacks;
 	/* pin callback routine enable flags, by pin number */
 	u32_t pin_callback_enables;
+	/* pin association with PINT id */
+	pint_pin_int_t pint_id[32];
+	/* interrupt type */
+	pint_pin_enable_t int_mode[32];
```

Comment:
I didn't dig into interrupt architecture much, however the `int_mode` doesn't seem to be used. There is one line which assigns a value to the field but there doesn't seem to be one that reads it. We are using here 32 bytes (or rather sizeof(pint_pin_enable_t)) of static RAM per port. We shouldn't if we don't have to.

Reply:
Will be removed, it's a remainder from porting to the new api and how interrupt is separate from general config.